pkg/network/ovn_kubernetes_test.go

```diff
@@ -70,6 +70,7 @@ func TestRenderOVNKubernetes(t *testing.T) {
 	g.Expect(objs).To(ContainElement(HaveKubernetesID("ClusterRoleBinding", "", "openshift-ovn-kubernetes-node")))
 	g.Expect(objs).To(ContainElement(HaveKubernetesID("DaemonSet", "openshift-ovn-kubernetes", "ovnkube-master")))
 	g.Expect(objs).To(ContainElement(HaveKubernetesID("DaemonSet", "openshift-ovn-kubernetes", "ovnkube-node")))
+	g.Expect(objs).To(ContainElement(HaveKubernetesID("ConfigMap", "openshift-ovn-kubernetes", "ovnkube-config")))
```

Comment:
I think we need more than this... You should be able to extract the generated `ovnkube.conf` with:

```
func extractOVNKubeConfig(objs []*uns.Unstructured) string {
	for _, obj := range objs {
		if obj.GetKind() == "ConfigMap" && obj.GetName() == "ovnkube-config" {
			val, ok, err := uns.NestedString(obj.Object, "data", "ovnkube.conf")
			g.Expect(err).NotTo(HaveOccurred())
			g.Expect(ok).To(BeTrue())
			return string(val)
		}
	}
}
```

So, create a new test function, starting with the top of `TestRenderOVNKubernetes`. After calling `renderOVNKubernetes`, do `conffile := extractOVNKubeConfig(objs)` to get the generated `ovnkube.conf` file as a string and then you can scan for expected / not-expected substrings in that. You can do it a few times, modifying `config` and re-running `renderOVNKubernetes` on it to check different results. (`TestProxyArgs` in `openshift_sdn_test.go` is the same idea, although in that case the config file is YAML so it uses `uns.Unstructured` to parse it rather than just treating it as a string.)

Reply:
added some more test cases